Sources/Tracking/CustomerIOImplementation.swift

```diff
@@ -208,28 +202,21 @@ class CustomerIOImplementation: CustomerIOInstance {
      is no active customer, this will fail to register the device
      */
     public func registerDeviceToken(_ deviceToken: String) {
-        addDeviceAttributes(deviceToken: deviceToken)
-    }
+        logger.info("initiating device token registration for \(deviceToken)")
+        // if the request is not forwarded to DataPipeline, make sure to save the device token for later use.
+        // so if a customer is identified later, we can reference the token and register it to a new profile.
 
-    /**
-     Adds device default and custom attributes and registers device token.
-     */
-    // TODO: Segment doesn't provide this method by default needs to get added
-    private func addDeviceAttributes(deviceToken: String, customAttributes: [String: Any] = [:]) {
-        // TODO: add support for device attributes in DataPipeline
-        DataPipeline.shared.registerDeviceToken(deviceToken)
-
-        logger.info("registering device token \(deviceToken)")
-        logger.debug("storing device token to device storage \(deviceToken)")
-        // no matter what, save the device token for use later. if a customer is identified later,
-        // we can reference the token and register it to a new profile.
-        globalDataStore.pushDeviceToken = deviceToken
+        guard let identifier = DataPipeline.shared.analytics.userId else {
+            logger.debug("storing device token to device storage \(deviceToken)")
+            globalDataStore.pushDeviceToken = deviceToken
```

Comment:
` DataPipeline.shared.registerDeviceToken(deviceToken)` already does this, but couldn't think of better solution for early returns at the moment

Reply:
Removed this as we decided to remove more checks